components/dialog/dialog-mixin.js

```diff
@@ -453,6 +454,8 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			else await animPromise;
 
 			/** Dispatched when the dialog is opened */
+			await this.#waitForUpdateComplete();
```

Comment:
I moved this to the mixin instead of in `<d2l-dialog>` directly, since I don't think we want the behaviour to be limited to just `async` dialog.